File: JavascriptAlgorithm/Tree/postorderTraversal-145.js
var postorderTraversal = function (root) {
  // left right root
  let res = [];
  function traverse(node) {
    if (node === null) {
      return;
    }
    traverse(node.left);
    traverse(node.right);
    res.push(node.val);
  }
  traverse(root);
  return res;
};
export { postorderTraversal };

File: Clion/ZeroTrac/Tree/postorderTraversal_145.cpp
//
// Created by TzashiNorpu on 10/31/2022.
//
#include "tree.h"

using namespace Tree;

vector<int> Solution::postorderTraversal(TreeNode *root) {
  vector<int> res;
  function<void(TreeNode *)> helper = [&](TreeNode *node) -> void {
      if (!node) return;
      helper(node->left);
      helper(node->right);
      res.push_back(node->val);
  };
  helper(root);
  return res;
}